acme/client.go

```diff
@@ -408,9 +408,11 @@ func (c *Client) solveChallenges(challenges []authorizationResource) map[string]
 			for i, solver := range solvers {
 				// TODO: do not immediately fail if one domain fails to validate.
 				err := solver.Solve(authz.Body.Challenges[i], authz.Domain)
-				if err != nil {
-					failures[authz.Domain] = err
+				if err == nil {
+					// we're done
+					return nil
```

Comment:
Hmm, that's possible. I think/thought that the loop for going through each domain was higher up in the code, but maybe not. I was only testing it with one domain.

With the current behavior any failure will make it fail, as far as I could tell.